include/aikido/planner/vectorfield/VectorFieldIntegrator.hpp

```diff
@@ -0,0 +1,107 @@
+#ifndef AIKIDO_PLANNER_VECTORFIELD_VECTORFIELDINTEGRATOR_HPP_
+#define AIKIDO_PLANNER_VECTORFIELD_VECTORFIELDINTEGRATOR_HPP_
+
+#include <dart/common/Timer.hpp>
+#include <aikido/constraint/Testable.hpp>
+#include <aikido/planner/PlanningResult.hpp>
+#include <aikido/planner/vectorfield/VectorField.hpp>
+#include <aikido/planner/vectorfield/VectorFieldPlannerStatus.hpp>
+#include <aikido/statespace/dart/MetaSkeletonStateSpace.hpp>
+#include <aikido/trajectory/Spline.hpp>
+
+namespace aikido {
+namespace planner {
+namespace vectorfield {
+
+struct Knot
+{
+  /// Timestamp.
+  double mT;
+
+  /// Positions.
+  Eigen::VectorXd mPositions;
+};
+
+/// VectorField Planner generates a trajectory by following a vector field
+/// defined in joint space.
+/// A planned trajectroy ends either by a predefined termination criterion or a
+/// integral time.
+///
+/// This class defines two callback functions for a integrator.
+/// step() provides joint velocities the vector field planner should follow,
+/// check() is called after each integration step to check planner status.
+class VectorFieldIntegrator
```

Comment:
I still don't understand why this is a class. Like I said before, I don't see any reason why you would ever want to call `followVectorField()` more than once on an instance of this class.

I suggest pulling `followVectorField()` out into a free function. If we need to maintain state between integration steps, e.g. the partially-complete trajectory, then we can instantiate a `VectorFieldIntegrator` class inside that function. However, that class is an *implementation detail* that belongs *in the `.cpp` file*.